services/etcd.go

```diff
@@ -373,3 +414,14 @@ func RestoreEtcdSnapshot(ctx context.Context, etcdHost *hosts.Host, prsMap map[s
 	}
 	return docker.RemoveContainer(ctx, etcdHost.DClient, etcdHost.Address, EtcdRestoreContainerName)
 }
+
+func configS3BackupImgCmd(ctx context.Context, imageCfg *container.Config, s3Backend *v3.S3BackupBackend) *container.Config {
+	log.Infof(ctx, "invoking s3 backup server cmd config, bucketName:%s, endpoint:%s", s3Backend.BucketName, s3Backend.Endpoint)
+	imageCfg.Cmd = append(imageCfg.Cmd, "--s3-backup=true")
```

Comment:
I think it would be easier if you group these in a single slice first:
```
cmd := []string{
"--s3-backup=true",
--s3-endpoint="+s3Backend.Endpoint,
.
.
"--s3-region="+s3Backend.Region,
}
return append(imageCfg.Cmd, cmd...)
```